lib/beaker/dsl/install_utils/pe_utils.rb

```diff
@@ -361,9 +361,12 @@ def do_install hosts, opts = {}
             pre30master = version_is_less(opts[:pe_ver] || master['pe_ver'], '3.0')
           end
 
+          agent_only_check_needed, hosts_agent_only, hosts_not_agent_only = \
+            create_hosts_agent_only_specified_arrays_helper(hosts, opts['pe_ver'])
```

Comment:
opts['pe_ver'] could be null if the user is setting the pe_ver per-host instead of globally - then I don't believe that the logic here works correctly.